Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,259 @@
+# Regular Expression Literals
+
+- Authors: Hamish Knight, Michael Ilseman
+
+## Introduction
+
+We propose to introduce a first-class regular expression literal into the language that can take advantage of library support to offer extensible, powerful, and familiar textual pattern matching.
+
+This is a component of a larger string processing picture. We would like to start a focused discussion surrounding our approach to the literal itself, while acknowledging that evaluating the utility of the literal will ultimately depend on the whole picture (e.g. supporting API). To aid this focused discussion, details such as the representation of captures in the type system, semantic details, extensions to lexing/parsing, additional API, etc., are out of scope of this pitch and thread. Feel free to continue discussion of anything related in the [overview thread][overview].
+
+## Motivation
+
+Regular expressions are a ubiquitous, familiar, and concise syntax for matching and extracting text that satisfies a particular pattern. Syntactically, a regex literal in Swift should:
+
+- Be familiar and facilitate knowledge reuse
+- Fit in with Swift's library extensibility story for literals
```

Comment:
```suggestion
- Be familiar and facilitate knowledge reuse
- Facilitate some reuse of existing regexs (e.g. found on StackOverflow)
- Fit in with Swift's library extensibility story for literals
```

I think we could add one more bullet here about facilitating re-use of actual regexs and not just knowledge... also 3 is the magic number for number of bullets.

Reply:
Agreed, these goals from @beccadax are great!